sql/resource_alert.go

```diff
@@ -0,0 +1,121 @@
+	"github.com/databricks/terraform-provider-databricks/common"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+)
+
+func ResourceAlert() common.Resource {
+	s := common.StructToSchema(sql.Alert{}, func(m map[string]*schema.Schema) map[string]*schema.Schema {
+		common.CustomizeSchemaPath(m, "display_name").SetRequired()
+		common.CustomizeSchemaPath(m, "query_id").SetRequired()
+		common.CustomizeSchemaPath(m, "condition").SetRequired()
+		// TODO: can we automatically generate it from SDK? Or should we avoid validation at all?
+		common.CustomizeSchemaPath(m, "condition", "op").SetRequired().SetValidateFunc(validation.StringInSlice([]string{
+			"GREATER_THAN", "GREATER_THAN_OR_EQUAL", "LESS_THAN", "LESS_THAN_OR_EQUAL", "EQUAL", "NOT_EQUAL", "IS_NULL"}, true))
+		common.CustomizeSchemaPath(m, "condition", "op").SetRequired()
+		common.CustomizeSchemaPath(m, "parent_path").SetCustomSuppressDiff(common.WorkspaceOrEmptyPathPrefixDiffSuppress).SetForceNew()
+		common.CustomizeSchemaPath(m, "condition", "operand").SetRequired()
+		common.CustomizeSchemaPath(m, "condition", "operand", "column").SetRequired()
+		common.CustomizeSchemaPath(m, "condition", "operand", "column", "name").SetRequired()
+		common.CustomizeSchemaPath(m, "condition", "empty_result_state").SetValidateFunc(
+			validation.StringInSlice([]string{"UNKNOWN", "OK", "TRIGGERED"}, true))
+		// We may not need it for some conditions
+		// common.CustomizeSchemaPath(m, "condition", "threshold").SetRequired()
+		common.CustomizeSchemaPath(m, "condition", "threshold", "value").SetRequired()
+		alof := []string{
+			"condition.0.threshold.0.value.0.string_value",
+			"condition.0.threshold.0.value.0.double_value",
+			"condition.0.threshold.0.value.0.bool_value",
+		}
+		for _, f := range alof {
+			common.CustomizeSchemaPath(m, "condition", "threshold", "value",
+				strings.TrimPrefix(f, "condition.0.threshold.0.value.0.")).SetAtLeastOneOf(alof)
+		}
+		common.CustomizeSchemaPath(m, "owner_user_name").SetSuppressDiff()
+		common.CustomizeSchemaPath(m, "id").SetReadOnly()
+		common.CustomizeSchemaPath(m, "create_time").SetReadOnly()
+		common.CustomizeSchemaPath(m, "lifecycle_state").SetReadOnly()
+		common.CustomizeSchemaPath(m, "state").SetReadOnly()
+		common.CustomizeSchemaPath(m, "trigger_time").SetReadOnly()
+		common.CustomizeSchemaPath(m, "update_time").SetReadOnly()
+		return m
+	})
+
+	return common.Resource{
+		Create: func(ctx context.Context, d *schema.ResourceData, c *common.DatabricksClient) error {
+			w, err := c.WorkspaceClient()
+			if err != nil {
+				return err
+			}
+			var a sql.CreateAlertRequestAlert
+			common.DataToStructPointer(d, s, &a)
+			apiAlert, err := w.Alerts.Create(ctx, sql.CreateAlertRequest{
+				Alert: &a,
+			})
+			if err != nil {
+				return err
+			}
+			d.SetId(apiAlert.Id)
+			owner := d.Get("owner_user_name").(string)
```

Comment:
We can't set Owner on create? :(

Reply:
No, I raised it to the Redash team, but they don't support it. We have same situation with UC resources